Update docs for multi rest values (#27537)
home-assistant/home-assistant.github.io: source/_integrations/sensor.rest.markdown

{% raw %}

```yaml
rest:

>   - resource: http://date.jsontest.com/

Reviewer: This needs to be quoted, as it is a string.

See: <https://developers.home-assistant.io/docs/documenting/yaml-style-guide#strings>

Author: https://github.com/home-assistant/home-assistant.io/pull/27593

{% raw %}

```yaml
rest:

>   - resource: https://api.openweathermap.org/data/2.5/weather?zip=80302,us&APPID=VERYSECRETAPIKEY

Reviewer: Same as above

Author: https://github.com/home-assistant/home-assistant.io/pull/27593

    sensor:
      - name: "Report"
        value_template: "{{ value_json['weather'][0]['description'].title() }}"
        picture: "{{ 'https://openweathermap.org/img/w/' + value_json['weather'][0]['icon'].lower() + '.png' }}"

      - name: "Outside temp"
        value_template: "{{ value_json['main']['temp'] - 273.15 }}"
        unit_of_measurement: "°C"

      - name: "Outside pressure"
        value_template: "{{ value_json['main']['pressure'] }}"
        unit_of_measurement: "hP"

      - name: "Outside humidity"
        value_template: "{{ value_json['main']['humidity'] }}"
        unit_of_measurement: "%"
```

{% endraw %}

> This configuration shows how to extract multiple values from a dictionary. This avoids flooding the REST service by only requesting the result once, then creating multiple sensors from that single result using template sensors.

Reviewer 2: ```suggestion
This configuration shows how to extract multiple values from a dictionary. This method avoids flooding the REST service because the result is only requested once. From that single request, multiple sensors can be created by using template sensors.
```

{% raw %}

```yaml
rest:

>     resource: http://<address_to_rest_service>

Reviewer: same

Author: https://github.com/home-assistant/home-assistant.io/pull/27593

    sensor:
      - name: "Bedroom1 Temperature"
        value_template: "{{ value_json['bedroom1']['temperature'] }}"
        device_class: temperature
        unit_of_measurement: "°C"
      - name: "Bedroom1 Humidity"
        value_template: "{{ value_json['bedroom1']['humidity'] }}"
        device_class: humidity
        unit_of_measurement: "%"
      - name: "Bedroom1 Battery"
        value_template: "{{ value_json['bedroom1']['battery'] }}"
        device_class: battery
        unit_of_measurement: "V"
      - name: "Bedroom2 Temperature"
        value_template: "{{ value_json['bedroom2']['temperature'] }}"
        device_class: temperature
        unit_of_measurement: "°C"
```

{% endraw %}

> The below example allows shows how to extract multiple values from a dictionary from the XML of a Steamist Steambath Wi-Fi interface and use them to create a switch and multiple sensors without having to poll the endpoint numerous times.

Reviewer 2: ```suggestion
The example below shows how to extract multiple values from a dictionary from the XML file of a Steamist Steambath Wi-Fi interface. The values are used to create a switch and multiple sensors without having to poll the endpoint numerous times.
```

{% raw %}

```yaml
rest:
  # Steam Controller

>   - resource: http://192.168.1.105/status.xml

Reviewer: idem

Author: https://github.com/home-assistant/home-assistant.io/pull/27593

    scan_interval: 15
  
    sensor:

>       - name: Steam Temp

Reviewer: idem

Author: https://github.com/home-assistant/home-assistant.io/pull/27593
